src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -118,6 +118,13 @@ public Expression Bind(ICSharpBinder payload, Expression[] parameters, DynamicMe
 
         #region Helpers
 
+        [ConditionalAttribute("DEBUG")]
+        internal static void EnsureLockIsTaken()
+        {
+            // Make sure that the binder lock is taken
+            System.Threading.Monitor.IsEntered(s_bindLock);
```

Comment:
This is a nop.  You need a Debug.Assert for the result of this.

Reply:
For some reason I though the IsEntered will throw :)